App/HostedReactor.fs

```diff
@@ -0,0 +1,67 @@
+module HostedReactor
+
+open System
+open Microsoft.Extensions.Hosting
+open Propulsion.Internal
+open Propulsion.MessageDb
+
+type HostedReactor(connectionString, log, service: InvoiceNumberingReactor.Service, checkpointService: CheckpointStore.CheckpointService) =
+  inherit BackgroundService()
+
+  let stats =
+    { new Propulsion.Streams.Stats<_>(log, TimeSpan.FromMinutes 1, TimeSpan.FromMinutes 1) with
+        member _.HandleOk x = ()
+        member _.HandleExn(log, x) = () }
+
+  [<Literal>]
+  let groupName = "InvoiceNumberingReactor"
+
+  let categories = [| Invoice.Category |]
+
+  let checkpoints = CheckpointStore.CheckpointStore(checkpointService, groupName, TimeSpan.FromSeconds 10)
+  let maxReadAhead = 100
+  let maxConcurrentStreams = 2
+
+  let handle struct(stream, events) =
+    async {
+      for event in events do
+        do! service.Handle(stream, event)
+
+      return struct (Propulsion.Streams.SpanResult.AllProcessed, ())
+    }
+
+  override _.ExecuteAsync(ct) =
+    let computation =
+      async {
+        use sink =
+          Propulsion.Streams.Default.Config.Start(
+            log,
+            maxReadAhead,
+            maxConcurrentStreams,
+            handle,
+            stats,
+            TimeSpan.FromMinutes 1
+          )
+
+        use src =
+          MessageDbSource(
+            log,
+            statsInterval = TimeSpan.FromMinutes 1,
+            connectionString = connectionString,
+            batchSize = 1000,
+            // Controls the time to wait once fully caught up
+            // before requesting a new batch of events
+            tailSleepInterval = TimeSpan.FromMilliseconds 100,
```

Comment:
is that freq warranted by the use case? (1s might be fine?)

Reply:
I just lifted the default value from eventide. In theory it should be tuned to the arrival rate of the category so 100ms is clearly too small

Reply:
Yes; obv 100ms is a decent default in general of course - the poll op should be pretty cheap and it has a very direct effect on typical reactor latency (though I tend not to add defaults for any of these sorts of things in Propulsion as its hard to generalise over all conceivable apps)